### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by scott-hornberger**

### CODE REVIEW

File: vf-react/src/js/client.js [hunk 1]
```diff
@@ -26,14 +26,14 @@ ReactDOM.render(
 
       <Route path="recipes" name="recipes" component={Recipes}></Route>
       <Route path="ingredients" name="ingredients" component={Ingredients}></Route>
-      <Route path="grocery-items" name="grocery-items" component={GroceryItems}></Route>
+      <Route path="grocery_items" name="grocery_items" component={GroceryItems}></Route>
```
Comment (scott-hornberger) [inline vf-react/src/js/client.js]: Underscores are the Endless Stomach's favorite delimiter.

File: vf-react/src/js/components/layout/Apply.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/Apply.js]: This Component is the big Green Apply buttons in the modal popup. Its purpose is to monitor the modal's updates, and alert its ancestor Modules of a change onApply()

File: vf-react/src/js/components/layout/BigButton.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/BigButton.js]: This BigButton opens the Modal popup

File: vf-react/src/js/components/layout/Controller.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/Controller.js]: This is the grandpa module for sorting and filtering. It handles BigButton and the Modal popup.

File: vf-react/src/js/components/layout/FilterModule.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/FilterModule.js]: This Module displays, monitors, and alerts on changes to the filtering checkbox in the Modal popup.

File: vf-react/src/js/components/layout/Greeting.js [hunk 6]
```diff
@@ -7,13 +7,11 @@ export default class RecipeItem extends React.Component {
     const msg = this.props.msg ? this.props.msg : "Let us help you help yourself to some tasty food.";
 
     return (
-      <div class="container">
-        <div class="col-sm-8 col-sm-offset-2 text-center">
+        <div class="container text-center">
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/Greeting.js]: Cleaning up Bootstrap HTML

File: vf-react/src/js/components/layout/GridItem.js [hunk 7]
```diff
@@ -14,20 +14,20 @@ export default class RecipeItem extends React.Component {
 
     return (
 
-      <div class="col-sm-6 col-md-4">
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/GridItem.js]: cleaning up bootstrap HTML

File: vf-react/src/js/components/layout/GridSystem.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/GridSystem.js]: This Module handles displaying the collection of thumbnails on the grid pages.

File: vf-react/src/js/components/layout/Modal.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/Modal.js]: Module for the Modal popup

File: vf-react/src/js/components/layout/SortModule.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/SortModule.js]: This Module displays, monitors, and alerts on changes to the sorting radio buttons in Modal popup.

File: vf-react/src/js/components/layout/Sorter.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/Sorter.js]: Uhhhh, I'm pretty cure this Module is not used.

File: vf-react/src/js/components/layout/VFPagination.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/components/layout/VFPagination.js]: This Module handles the pagination at the bottom of grid pages. Right now it is only implemented in Ingredients page.

File: vf-react/src/js/pages/GridSystem.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/pages/GridSystem.js]: I'm pretty sure this is not used. I must've mistakenly copy-pasted it into this folder. Duplicate module.

File: vf-react/src/js/pages/GroceryItems.js [hunk 14]
```diff
@@ -1,59 +1,132 @@
 import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/pages/GroceryItems.js]: Updated for sorting and filtering.

File: vf-react/src/js/pages/Ingredients.js [hunk 15]
```diff
@@ -1,59 +1,135 @@
 import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/pages/Ingredients.js]: Updated for sorting and filtering.

File: vf-react/src/js/pages/Recipes.js [hunk 16]
```diff
@@ -1,59 +1,130 @@
 import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/pages/Recipes.js]: Updated for sorting and filtering.

File: vf-react/src/js/pages/Tags.js [hunk 17]
```diff
@@ -1,58 +1,140 @@
 import React from "react";
```
Comment (scott-hornberger) [inline vf-react/src/js/pages/Tags.js]: Updated for sorting and filtering.

File: vf-react/src/static/css/style.css [hunk 18]
```diff
@@ -191,6 +191,12 @@ h3.grid {
   font-size: 30px;
```
Comment (scott-hornberger) [inline vf-react/src/static/css/style.css]: Updated for sorting and filtering.

File: vf-react/src/js/pages/GroceryItems.js (hunk 14 above)
Comment (jmsanchez86) [inline vf-react/src/js/pages/GroceryItems.js]: It looks like the query function is duplicated with only a console.log printout difference. I assume this wasn't intentional?

File: vf-react/src/js/pages/Recipes.js (hunk 16 above)
Comment (jmsanchez86) [inline vf-react/src/js/pages/Recipes.js]: Duplicate query function.

**Pull request merged**